src/osm-gps-map/osm-gps-map-qt.cpp

```diff
@@ -218,7 +219,10 @@ Maep::GpsMap::GpsMap(QQuickItem *parent)
   gint width = gconf_get_int(GCONF_KEY_TRACK_WIDTH, -1);
   bool orientation = gconf_get_bool(GCONF_KEY_SCREEN_ROTATE, TRUE);
   int gpsRefresh = gconf_get_int(GCONF_KEY_GPS_REFRESH_RATE, 1000);
-  bool compassEnabled = gconf_get_bool(GCONF_KEY_COMPASS_ENABLED, FALSE);
```

Comment:
Can you keep this line for backwrad compatibility and test if compassEnabled is true to MODE_NORTH ?